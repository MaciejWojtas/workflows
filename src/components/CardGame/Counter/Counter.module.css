.counter {
  @apply w-44 h-28 bg-black border-sun flex flex-col justify-between pt-2;
  border-width: 1px;
}

.counter__number {
  @apply text-white text-5xl;
}

.counter__text {
  @apply text-white;
}

.counter-aces {
  @apply h-6 mb-2;
}

.counter-aces__ace {
  @apply text-white mx-1;
}
