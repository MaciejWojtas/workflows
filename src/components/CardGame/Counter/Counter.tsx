import React, { FunctionComponent } from 'react';
import styles from './Counter.module.css';

type CardGameProps = {
  cardsLeft: number;
  acesLeft: number;
};

const Counter: FunctionComponent<CardGameProps> = ({ cardsLeft, acesLeft }) => {
  return (
    <div className={styles.counter}>
      <span className={`${styles.counter__number} font-secondary`}>{cardsLeft}</span>
      <span className={`${styles.counter__text} font-secondary`}>Cards Left</span>
      <div className={`${styles['counter-aces']} font-secondary`}>
        {[...Array(acesLeft)].map((_, i) => (
          <span className={styles['counter-aces__ace']} key={i}>
            A
          </span>
        ))}
      </div>
    </div>
  );
};

export default Counter;
