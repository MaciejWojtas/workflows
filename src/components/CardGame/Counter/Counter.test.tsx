import React from 'react';
import renderer from 'react-test-renderer';
import Counter from './Counter';

it('renders correctly without class name', () => {
  const tree = renderer.create(<Counter cardsLeft={2} acesLeft={4} />).toJSON();

  expect(tree).toMatchSnapshot();
});
