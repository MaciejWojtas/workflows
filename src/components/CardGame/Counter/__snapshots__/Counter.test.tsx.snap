// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly without class name 1`] = `
<div
  className="counter"
>
  <span
    className="counter__number font-secondary"
  >
    2
  </span>
  <span
    className="counter__text font-secondary"
  >
    Cards Left
  </span>
  <div
    className="counter-aces font-secondary"
  >
    <span
      className="counter-aces__ace"
    >
      A
    </span>
    <span
      className="counter-aces__ace"
    >
      A
    </span>
    <span
      className="counter-aces__ace"
    >
      A
    </span>
    <span
      className="counter-aces__ace"
    >
      A
    </span>
  </div>
</div>
`;
